compiler/src/dotty/tools/dotc/config/SourceVersion.scala

```diff
@@ -16,7 +16,10 @@ enum SourceVersion:
   def isAtLeast(v: SourceVersion) = stable.ordinal >= v.ordinal
 
 object SourceVersion extends Property.Key[SourceVersion]:
-  def defaultSourceVersion = `3.0`
+  def defaultSourceVersion = `3.0`  // should be 3.2? but some tests fail:
```

Comment:
if you rebase off from https://github.com/lampepfl/dotty/pull/14629 then these still pass with 3.2 as default